src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -226,24 +227,45 @@ object HttpGoogleServicesDAO extends FireCloudRequestBuilding {
             objectResponse.status match {
               case OK =>
                 // user has access to the object.
-                // now make a final request to see if our service account has access, so it can sign a URL
-                objectAccessCheck(bucketName, objectKey, getRawlsServiceAccountAccessToken) map { serviceAccountResponse =>
-                  serviceAccountResponse.status match {
-                    case OK =>
-                      // the service account can read the object too. We are safe to sign a url.
-                      log.info(s"$userStr download via signed URL allowed for [$objectStr]")
-                      requestContext.redirect(getSignedUrl(bucketName, objectKey), StatusCodes.TemporaryRedirect)
-                    case _ =>
-                      // the service account cannot read the object, even though the user can. We cannot
-                      // make a signed url, because the service account won't have permission to sign it.
-                      // therefore, we rely on a direct link. We accept that a direct link is vulnerable to
-                      // identity problems if the current user is signed in to multiple google identies in
-                      // the same browser profile, but this is the best we can do.
-                      // generate direct link per https://cloud.google.com/storage/docs/authentication#cookieauth
-                      log.info(s"$userStr download via direct link allowed for [$objectStr]")
-                      requestContext.redirect(getDirectDownloadUrl(bucketName, objectKey), StatusCodes.TemporaryRedirect)
+                // switch solutions based on the size of the target object. If the target object is small enough,
+                // proxy it through orchestration; this allows embedded images inside HTML reports to render correctly.
+                val objSize:Int = Try(objectResponse.entity.asString.parseJson.convertTo[ObjectMetadata].size)
+                                                    .toOption.getOrElse("-1").toInt
```

Comment:
It means google changed their response - which shouldn't happen. It's pretty defensive coding, but ... you know google!